runners/flink/src/main/java/org/apache/beam/runners/flink/metrics/FlinkMetricContainer.java

```diff
@@ -62,23 +62,28 @@
   private final Map<String, FlinkGauge> flinkGaugeCache;
   private final MetricsAccumulator metricsAccumulator;
 
-  public FlinkMetricContainer(RuntimeContext runtimeContext) {
+  public FlinkMetricContainer(RuntimeContext runtimeContext, boolean useAccumulator) {
     this.runtimeContext = runtimeContext;
     this.flinkCounterCache = new HashMap<>();
     this.flinkDistributionGaugeCache = new HashMap<>();
     this.flinkGaugeCache = new HashMap<>();
 
-    Accumulator<MetricsContainerStepMap, MetricsContainerStepMap> metricsAccumulator =
-        runtimeContext.getAccumulator(ACCUMULATOR_NAME);
-    if (metricsAccumulator == null) {
-      metricsAccumulator = new MetricsAccumulator();
-      try {
-        runtimeContext.addAccumulator(ACCUMULATOR_NAME, metricsAccumulator);
-      } catch (UnsupportedOperationException e) {
-        // Not supported in all environments, e.g. tests
-      } catch (Exception e) {
-        LOG.error("Failed to create metrics accumulator.", e);
+    Accumulator<MetricsContainerStepMap, MetricsContainerStepMap> metricsAccumulator;
```

Comment:
```suggestion
    final Accumulator<MetricsContainerStepMap, MetricsContainerStepMap> metricsAccumulator;
```

Reply:
That doesn't work because the accumulator can be assigned multiple times. See the code below. (I don't want to use another variable)